src/app_desc.rs

```diff
@@ -0,0 +1,46 @@
+
+pub const fn strToCharArray<const C: usize>(s: &str) -> [i8; C] {
+    let mut ret = [0i8; C];
+    let mut i = 0;
+
+    while i < C {
+        if i < s.len() {
+            ret[i] = s.as_bytes()[i] as i8;
+        } else {
+            break;
+        }
+
+        i += 1;
+    }
+
+    ret
+}
+
+#[macro_export]
+macro_rules! esp_app_desc {
+    () => {
+        use build_time::build_time_utc;
```

Comment:
I don't see the change to `Cargo.toml` where `esp-idf-sys` now depends on the `build_time` and `const_format` crates.

Reply:
That's the problem. It doesn't. The macro gets compiles on the user create, do the *user* must depend on it. I tried adding it to esp-idf-sys, but it won't do, because the user create - where the macro is expanded to - doesn't see it, unless added to it's own cargo.toml... Cargo hides transitive dependencies unless they are explicit

Reply:
See the last comments by me and @N3xed in the general issue discussion.